src/Common/src/System/IO/PathInternal.cs

```diff
@@ -10,22 +10,6 @@ namespace System.IO
     /// <summary>Contains internal path helpers that are shared between many projects.</summary>
     internal static partial class PathInternal
     {
-        /// <summary>
-        /// Checks for invalid path characters in the given path.
-        /// </summary>
-        /// <exception cref="System.ArgumentNullException">Thrown if the path is null.</exception>
-        /// <exception cref="System.ArgumentException">Thrown if the path has invalid characters.</exception>
-        /// <param name="path">The path to check for invalid characters.</param>
-        internal static void CheckInvalidPathChars(string path)
-        {
-            if (path == null)
-                throw new ArgumentNullException(nameof(path));
-
-            if (HasIllegalCharacters(path))
```

Comment:
@JeremyKuhne, what about direct call sites to HasIllegalCharacters, e.g. https://github.com/dotnet/corefx/blob/0eb5e7451028e9374b8bb03972aa945c128193e1/src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Windows.cs#L25-L26 ?

Reply:
These should be removed too.

Reply:
should I remove the function  HasIllegalCharacters  from the corefx as it is not being used anywhere else

Reply:
Yes, please go ahead and remove any dead methods.

Reply:
<test name="System.IO.FileSystem.DriveInfoTests.DriveInfoWindowsTests.Ctor_InvalidPath_ThrowsArgumentException(driveName: \&quot;:\\0\&quot;, paramName: \&quot;driveName\&quot;)" type="System.IO.FileSystem.DriveInfoTests.DriveInfoWindowsTests" method="Ctor_InvalidPath_ThrowsArgumentException" time="0.0033929" result="Fail">
        <failure exception-type="Xunit.Sdk.EqualException">
          <message><![CDATA[Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: driveName\r\nActual:   path\r\n          ↑ (pos 0)]]></message>
          <stack-trace><![CDATA[   at System.AssertExtensions.Throws[T](String netCoreParamName, String netFxParamName, Action action) in c:\git\corefx3\src\CoreFx.Private.TestUtilities\src\System\AssertExtensions.cs:line 39
   at System.IO.FileSystem.DriveInfoTests.DriveInfoWindowsTests.Ctor_InvalidPath_ThrowsArgumentException(String driveName, String paramName) in c:\git\corefx3\src\System.IO.FileSystem.DriveInfo\tests\DriveInfo.Windows.Tests.cs:line 36]]></stack-trace>
        </failure>
      </test>

@JeremyKuhne 
this is the disabled test that failed after removing the check for hasIllegalCharacters. should I remove\disable the test or keep this check.

Reply:
You can remove the ones that hit the invalid char checks. There is also a check for proper formatting of the `C:`, so some of these are still valid.